### PR TITLE
refactor: migrate NotificationIcon calls to eloquent ORM

### PR DESCRIPTION
**Pull request by wescopeland**
I'm trying to write tests for https://github.com/RetroAchievements/RAWeb/pull/1833. The page requires auth, and I encountered some errors when using `actingAs()` in my test class. I tracked these errors down to the notification icon using legacy mysqli calls.

These refactors allow me to use `actingAs()` in the test class I'm working on.

### CODE REVIEW

File: app/Helpers/database/set-claim.php [hunk 1]
```diff
@@ -455,26 +457,27 @@ function updateClaim(int $claimID, int $claimType, int $setType, int $status, in
...
-            COALESCE(SUM(CASE WHEN TIMESTAMPDIFF(MINUTE, NOW(), sc.Finished) BETWEEN 0 AND 10080 THEN 1 ELSE 0 END), 0) AS Expiring
-        FROM
-            SetClaim AS sc
-        WHERE
-            sc.User = '$user'
-            AND sc.Status IN (" . ClaimStatus::Active . ',' . ClaimStatus::InReview . ")
-            AND sc.Special != " . ClaimSpecial::ScheduledRelease;
-
-    $dbResult = s_mysql_query($query);
+    if (empty($user)) {
```
Comment (luchaos) [inline app/Helpers/database/set-claim.php:460]: here's an opportunity to rename this variable to `$username` to better describe what this is and not run into ambiguity with `$user` referencing a full user model

File: app/Helpers/database/set-claim.php [hunk 2]
```diff
@@ -455,26 +457,27 @@ function updateClaim(int $claimID, int $claimType, int $setType, int $status, in
...
-    if ($dbResult !== false) {
-        return mysqli_fetch_assoc($dbResult);
+    $claims = AchievementSetClaim::select(
+        DB::raw('COALESCE(SUM(CASE WHEN TIMESTAMPDIFF(MINUTE, NOW(), Finished) <= 0 THEN 1 ELSE 0 END), 0) AS Expired'),
+        DB::raw('COALESCE(SUM(CASE WHEN TIMESTAMPDIFF(MINUTE, NOW(), Finished) BETWEEN 0 AND 10080 THEN 1 ELSE 0 END), 0) AS Expiring')
+    )
+    ->where('User', $user)
+    ->whereIn('Status', [ClaimStatus::Active, ClaimStatus::InReview])
+    ->where('Special', '!=', ClaimSpecial::ScheduledRelease)
+    ->first();
```
Comment (luchaos) [inline app/Helpers/database/set-claim.php:471]: those usually would be indented by 4 spaces.

File: app/Helpers/database/ticket.php [hunk 3]
```diff
@@ -413,22 +415,14 @@ function countRequestTicketsByUser(string $user): int
...
-        WHERE ua.User = '$user' AND ach.Flags IN (3, 5) AND tick.ReportState = " . TicketState::Request;
-
-    $dbResult = s_mysql_query($query);
-
-    if (!$dbResult) {
-        return 0;
-    }
-
-    return (int) mysqli_fetch_assoc($dbResult)['count'];
+    return $count;
```
Comment (luchaos) [inline app/Helpers/database/ticket.php:425]: (indendation)
minor: I'd make this a direct return of the count without the local variable. we have that in some places but personally prefer to not use local variables when they do not add context for better understanding.

**Review by luchaos — APPROVED** [review]

**Comment by wescopeland** [comment]
All feedback resolved in https://github.com/RetroAchievements/RAWeb/pull/1836/commits/0e2e013ba388dfa375edf36b5bdc90ab93ba64ed.

**Pull request merged**